src/lib/cloud-provider.js

```diff
@@ -57,12 +58,31 @@ class CloudProvider {
     }
 
     onOpen () {
+        this.connectionAttempts = 1; // Reset because we successfully connected
         this.writeToServer('handshake');
         log.info(`Successfully connected to clouddata server.`);
     }
 
     onClose () {
         log.info(`Closed connection to websocket`);
+        const exponentialTimeout = (Math.pow(2, this.connectionAttempts) - 1) * 1000;
+        const randomizedTimeout = this.randomizeDuration(exponentialTimeout);
+        this.setTimeout(this.reconnectNow.bind(this), randomizedTimeout);
+    }
+
+    reconnectNow () {
+        // Max connection attempts at 5, so timeout will max out in range [0, 31s]
+        this.connectionAttempts = Math.min(this.connectionAttempts + 1, 5);
+        this.openConnection();
```

Comment:
should there be a check here that we haven't exceeded the connectionAttempts before opening the connection?

Reply:
Similar to comment above, the connection attempt number gets limited just to max the timeout, but it keeps reconnecting ad infinitum